### ISSUE TITLE
Support for "NaN" and "Inf" for floating point.

### ISSUE THREAD

**Issue by drug007** [tamediadigital/asdf#81]
Is it planned?

**Comment by 9il** [comment]
example,plz

**Comment by drug007** [comment]
```
unittest
{
	static struct S
	{
		static string static_string = "me shouldn't be serialized at all!";
		string str = "me shall be serialized";

		static @property staticProperty()
		{
			return static_string;
		}

		enum A = 200;
		const float B = 10f;
		float f;
		int i;

		bool opEquals()(auto ref const(typeof(this)) rhs)
		{
			import std.math : isNaN;

			if (f.isNaN && rhs.f.isNaN)
			{
				// if both f are nan then they are equal
			}
			else if (f != rhs.f)
				return false;

			return str == rhs.str && i == rhs.i;
		}
	}

	assert (S.staticProperty == S.static_string);

        // Asdf works with nan
	{
		auto asdf = serializeToAsdf(S());
                import std.conv : text;
		assert (asdf.text == `{"str":"me shall be serialized","B":10,"f":nan,"i":0}`);
		assert (S() == deserialize!S(asdf));
	}

        // Json doesn't work
	{
		auto json = serializeToJson(S());
		assert (json.text == `{"str":"me shall be serialized","B":10,"f":nan,"i":0}`);
		assert (S() == deserialize!S(json));
	}
}
```
parsing Json gives the run-time error:
```
asdf.asdf.AsdfException@source/asdf/jsonparser.d(76): unexpected character when parsing 'null'
...
```

**Comment by drug007** [comment]
Also difference between asdf and json are rather shadowy for me, probably I should use asdf instead of json?

**Comment by 9il** [comment]
Hmm, we need to add `(-)nan` and`(+)inf` to asdf standard. The should be serialised to json strings  like `"nan"`

**Comment by drug007** [comment]
IIRC Asdf already do it, but json grammar doesn't support infinity and not-a-number, it's extensions with compatibility problems. I guess it would be enough to serialize `{float|double}.{nan|inf}` as `null` in json and deserialize it as `{float|double}.nan` - we loose `inf` but it will be more compatible with other parsers. And in Asdf provide full support.

**Comment by drug007** [comment]
Closed via #82
